app/org/sagebionetworks/bridge/services/ConsentService.java

```diff
@@ -294,45 +340,78 @@ private void sendWithdrawEmail(Study study, String externalId, Account account,
 
+        boolean verifiedEmail = (participant.getEmail() != null
+                && Boolean.TRUE.equals(participant.getEmailVerified()));
+        boolean verifiedPhone = (participant.getPhone() != null
+                && Boolean.TRUE.equals(participant.getPhoneVerified()));
+        
         ConsentPdf consentPdf = new ConsentPdf(study, participant.getTimeZone(), participant.getEmail(),
                 consentSignature, sharingScope, studyConsentDocument, xmlTemplateWithSignatureBlock);            
         
-        BasicEmailProvider.Builder builder = new BasicEmailProvider.Builder()
-                .withStudy(study)
-                .withEmailTemplate(study.getSignedConsentTemplate())
-                .withBinaryAttachment("consent.pdf", MimeType.PDF, consentPdf.getBytes())
-                .withRecipientEmail(participant.getEmail());
-        addStudyConsentRecipients(study, builder);
-        // If the user doesn't have an email, we won't send anything. Note that as of this refactor, 
-        // we're not verifying the email address. Change this when SMS is added.
-        BasicEmailProvider provider = builder.build();
-        if (!provider.getRecipientEmails().isEmpty()) {
+        if (verifiedEmail) {
+            BasicEmailProvider provider = new BasicEmailProvider.Builder()
+                    .withStudy(study)
+                    .withEmailTemplate(study.getSignedConsentTemplate())
+                    .withBinaryAttachment("consent.pdf", MimeType.PDF, consentPdf.getBytes())
+                    .withRecipientEmail(participant.getEmail()).build();
             sendMailService.sendEmail(provider);
+        } else if (verifiedPhone) {
+            sendConsentViaSMS(study, subpop, participant, consentPdf);
+        } else {
+            throw new BadRequestException("Participant does not have a valid email address or phone number");
+        }
+    }
+    
+    private void sendConsentViaSMS(Study study, Subpopulation subpop, StudyParticipant participant,
+            ConsentPdf consentPdf) {
+        String shortUrl = null;
+        try {
+            String fileName = getSignedConsentUrl();
+            DateTime expiresOn = getDownloadExpiration();
+            s3Helper.writeBytesToS3(USERSIGNED_CONSENTS_BUCKET, fileName, consentPdf.getBytes());
+            URL url = s3Helper.generatePresignedUrl(USERSIGNED_CONSENTS_BUCKET, fileName, expiresOn, HttpMethod.GET);
+            shortUrl = urlShortenerService.shortenUrl(url.toString(), SIGNED_CONSENT_DOWNLOAD_EXPIRE_IN_SECONDS);
+        } catch(IOException e) {
+            throw new BridgeServiceException(e);
         }
+        SmsMessageProvider provider = new SmsMessageProvider.Builder()
+                .withStudy(study)
+                .withPhone(participant.getPhone())
+                .withExpirationPeriod(EXPIRATION_PERIOD_KEY, SIGNED_CONSENT_DOWNLOAD_EXPIRE_IN_SECONDS)
+                .withPromotionType()
```

Comment:
This should be Transactional, since this is a Governance requirement.

Reply:
Changed to transaction